capice.py

```diff
@@ -41,39 +47,39 @@ def main():
     verbose = cla.get_argument('verbose')
     force = cla.get_argument('force')
     train = cla.get_argument('train')
+    config_argument = convert_cla_to_str(cla.get_argument('config'))
+
+    # Initializing manager
+
+    manager = CapiceManager()
+    manager.config_loc = config_argument
+    manager.verbose = verbose
+    manager.force = force
 
     # Getting all config settings.
 
     config = ConfigReader()
     config.parse()
-    log_loc = config.get_default_value(key='logfilelocation')
 
     # Checking the input arguments.
 
     input_checker = InputChecker()
 
-    input_checker.check_input_loc(input_loc=input_loc)
-    input_checker.check_input_output_directories(input_path=input_loc, output_path=output_loc)
+    input_loc = input_checker.check_input_loc(input_loc=input_loc)
+    input_checker.check_input_output_directories(input_path=input_loc,
+                                                 output_path=output_loc)
     output_loc = input_checker.get_output_directory()
     output_filename = input_checker.get_output_filename()
     input_checker.check_output_loc(output_loc=output_loc)
 
-    # Checking the log arguments
+    # Setting the remaining manager variables.
 
-    lc = LogChecker(log_loc=log_loc, output_loc=output_loc)
-    log_loc = lc.check_log_loc()
-
-    # Initializing the manager
-
-    manager = CapiceManager()
-    manager.now = datetime.now()
-    manager.enable_logfile = config.get_misc_value(key='enablelogfile')
-    manager.log_loc = log_loc
-    manager.verbose = verbose
     manager.overwrite_impute = config.get_overwrite_value(key='imputefile')
     manager.overwrite_model = config.get_overwrite_value(key='modelfile')
-    manager.force = force
+    manager.config_vep_version = config.get_default_value(key='vepversion')
+    manager.config_grch_build = config.get_default_value(key='genomebuild')
     manager.output_filename = output_filename
+    manager.reference_genome = config.get_datafiles_value(key='reference')
```

Comment:
More of a design-technical point. Current flow before running main/train is:
1. Parse command line arguments
2. Retrieve & convert each argument individually (through using a separate utilities class)
3. Initializing the global manager (CapiceManager) & override some settings
4. Parse the config file
5. Retrieve & convert all items from the config file individually (through using separate checker class)
6. Override remaining settings of manager

Might it not be cleaner to do the following?
1. Run command line parser
2. Run config file parser
3. Initialize CapiceManager that requires these objects during it's initialization (or make the config file optional) and ensures everything is set accordingly & validates whether all required fields are set (or offers a method to call this externally that returns `true`/`false`). You could even remove the setters if preferred to indicate variables shouldn't be adjusted after it was initialized or replace the default setters with ones that only accept the command line/config file parser.

Reply:
Manager has to initialized before the config parser since the manager contains the possible location of a custom config file. 

If I make the CapiceManager require a certain amount of arguments upon initialization, i.e. `def __init__(self, *args)`, CapiceManager would require them each and every time I refer to the CapiceManager instance. I'm not sure how I can get past that.

Reply:
My main point is an alternative to assigning variables to each setting in `capice.py` and afterwards assigning them to the manager. You could simply retrieve that single argument from the command line parser and give it the the config parser (after having validated it), though the following would be more similar to the current flow:
1. Initialize empty CapiceManager
2. Run command line parser (decide on having custom getters that CapiceManager could use and it already validating its input on validity or this being done in CapiceManager)
3. run `CapiceManager.digestCmdParser(cmdParser)` (throws error if something went wrong)
4. run config parser (using a `CapiceManager.getConfigFile()`)
5. run `CapiceManager.digestConfigFile()` (throws error if something went wrong)
6. run `CapiceManager.valid()` or something like that returns true/false depending on whether everything that is required to run, has been set properly?

The `digest` functions could then also throw an error if something went wrong during digesting, ensuring a config file would f.e. be set & valid before a config file is trying to be digested.

Only downside is you cannot "ensure" `CapiceManager.getConfigFile()` has been set (getter could be called without first running `CapiceManager.digestCmdParser(cmdParser)`), though this would already reduce all these variables being set in `capice.py` which are actually meant for `CapiceManager`.

---

The reason I suggested using these as arguments during initialization was simply so that you could forgo the setters to indicate "none of these values should be changed afterwards". I guess with a Singleton you could have 2 constructors with the empty one throwing an error if none was set yet, but that also seems a bit unnecessary so perhaps the above example would be more fitting.

Reply:
Logic behind the reasoning: It would be cleaner if `capice.py` doesn't have to "know each individual variable" but can let this logic remain at the command line parser/config file parser/CapiceManager".

Reply:
Alternatively, you could simply apply the settings directly from within the cmdParser/configFile parser (and have the CapiceManager validate these when they get set).